lib/app.js

```diff
@@ -27,18 +27,9 @@ export default class App extends Component {
     }
   }
 
-  componentDidCatch (err, info) {
-    // To provide clearer stacktraces in error-debug.js in development
-    // To provide clearer stacktraces in app.js in production
-    err.info = info
-
-    if (process.env.NODE_ENV === 'production') {
-      // In production we render _error.js
-      window.next.renderError({err})
-    } else {
-      // In development we throw the error up to AppContainer from react-hot-loader
-      throw err
-    }
+  // Kept here for backwards compatibility.
+  componentDidCatch (err) {
+    throw err
```

Comment:
wish we could get rid of this, but oh well. Much cleaner 😍

Reply:
I'm not sure how we're going to deprecate this correctly, we can't warn since it's triggered in normal cases where an error is thrown too. I'm guessing we should just include it in the release notes / blogpost of Next 7.

Reply:
Doesn't `componentDidCatch` implicitly just throw err upwards anyway? I haven't used Error Boundaries, but from reading the release I understood that this works similar to `context` and it'll just bubble up.

Reply:
What's the purpose of the explicit componentDidCatch?

Reply:
It's backwards compatibility with the current implementation we told people to do this:

```
componentDidCatch(err, info) {
  console.log('YOUR OWN LOGGING')
  super.componentDidCatch(err, info)
}
```

Reply:
This is true when using a custom `pages/_app.js` that overrides this file and extends the App class.

Reply:
oh I see! They should also be able to:

Reply:
```

Reply:
With the new solution, it can be easily done "the React way" by just `throw`ing the error up the bubble.

Reply:
🎉